src/main/scala/tech/cryptonomic/conseil/michelson/parser/JsonParser.scala

```diff
@@ -0,0 +1,94 @@
+package tech.cryptonomic.conseil.michelson.parser
+
+import io.circe.parser.decode
+import cats.syntax.functor._
+import io.circe._
+import io.circe.generic.auto._
+import tech.cryptonomic.conseil.michelson.dto._
+
+class JsonParser {
+
+  sealed trait JsonSection
+
+  case class JsonTypeSection(prim: String, args: Option[Seq[JsonTypeSection]]) extends JsonSection {
+    def toMichelsonType: MichelsonType = MichelsonType(prim, args.getOrElse(Seq()).map(_.toMichelsonType))
```

Comment:
Actually, it's ``Option[List[]]`` so I would need flatMap which would complicate this code. I'm not sure if your proposition is quicker too because those operations are O(1) for None or empty List anyway.

Reply:
I find it easier to grasp, to be honest, not so much as for performance